grpc/shared_conn.go

```diff
@@ -194,16 +195,12 @@ func (sc *SharedConn) ResetConn(conn rpc.ClientConn, moduleLogger logging.Logger
 	}
 
 	sc.peerConn.OnTrack(func(trackRemote *webrtc.TrackRemote, rtpReceiver *webrtc.RTPReceiver) {
-		name, err := resource.NewFromString(trackRemote.StreamID())
-		if err != nil {
-			sc.logger.Errorw("StreamID did not parse as a ResourceName", "sharedConn", fmt.Sprintf("%p", sc), "streamID", trackRemote.StreamID())
-			return
-		}
-		sc.resOnTrackMu.Lock()
-		onTrackCB, ok := sc.resOnTrackCBs[name]
-		sc.resOnTrackMu.Unlock()
+		sc.onTrackCBByTrackNameMu.Lock()
```

Comment:
Removed the requirement that the streamID (aka track name) parse as a resource name as we are now just storing strings in the callback lookup map

Reply:
Is it worth it to have the onTrackByTrackName as a separate struct with a separate mutex within the client struct? I always worry with multiple mutexes on the same struct that we're making the code harder to handle.

Totally okay if the answer is no - too much redesign/I'm actually making it more complex.